### PR TITLE
Read units from listing

### PR DESCRIPTION
**Pull request by rap1ds**
- [X] Add ListingShape.update
- [X] Instead of reading unit from `listing.transaction_type.price_per`, use `listing.unit_type`

Merge after #1080

### CODE REVIEW

File: app/services/listing_service/store/shape.rb [hunk 1]
```diff
@@ -51,22 +62,67 @@ def create(community_id:, opts:)
...
+
     translations = opts[:translations] # Skip data type and validation, because this is temporary
 
-    # TODO We should be able to create transaction_type without community
-    community = Community.find(shape[:community_id])
+    tt_model = nil
+
+    ActiveRecord::Base.transaction do
+      # TODO We should be able to create transaction_type without community
+      community = Community.find(shape[:community_id])
```
Comment (ovan) [inline app/services/listing_service/store/shape.rb]: Use community_id attribute in TransactionType model directly instead.

File: app/services/listing_service/store/shape.rb [hunk 2]
```diff
@@ -51,22 +62,67 @@ def create(community_id:, opts:)
...
-    create_tt_opts = to_tt_model_attributes(shape).except(:units, :translations)
-    tt_model = community.transaction_types.build(create_tt_opts)
+  def update(community_id:, transaction_type_id: nil, listing_shape_id: nil, opts:)
+    tt_model = if transaction_type_id
+      TransactionTypeModel.where(community_id: community_id, id: transaction_type_id).first
+    elsif listing_shape_id
+      raise NotImplementedError.new("Can not find listing shape by listing_shape_id, yet. Specify transaction_type_id instead.")
+    else
+      raise ArgumentError.new("Can not find listing shape without id.")
+    end
```
Comment (ovan) [inline app/services/listing_service/store/shape.rb]: This block should go to separate (private) method

File: app/services/listing_service/store/shape.rb [hunk 3]
```diff
@@ -51,22 +62,67 @@ def create(community_id:, opts:)
...
 
-    units.each { |unit|
-      tt_model.listing_units.build(to_unit_model_attributes(unit))
-    }
-    translations.each { |tr| tt_model.translations.build(tr) }
+    translations = opts[:translations] # Skip data type and validation, because this is temporary
+
+    # TODO We should be able to create transaction_type without community
+    ActiveRecord::Base.transaction do
+      community = Community.find(community_id)
```
Comment (ovan) [inline app/services/listing_service/store/shape.rb]: This is not used

File: app/services/listing_service/store/shape.rb [hunk 4]
```diff
@@ -51,22 +62,67 @@ def create(community_id:, opts:)
...
+
+      update_tt_opts = HashUtils.compact(to_tt_model_attributes(update_shape)).except(:units, :translations)
+      tt_model.update_attributes(update_tt_opts)
+
+      unless units.nil?
+        tt_model.listing_units.destroy_all
+        units.each { |unit| tt_model.listing_units.build(to_unit_model_attributes(unit)) }
+      end
+
+      unless translations.nil?
```
Comment (ovan) [inline app/services/listing_service/store/shape.rb]: So update rebuilds all unit rows every time? This implies that units cannot ever be referenced from outside.

File: app/controllers/listings_controller.rb [hunk 5]
```diff
@@ -123,7 +123,7 @@ def show
       listing_id: @listing.id.to_s,
       payment_gateway: payment_gateway,
       payment_process: process,
-      booking: @listing.transaction_type.price_per.present?
+      booking: @listing.unit_type == :day
     )
```
Comment (ovan) [inline app/controllers/listings_controller.rb]: Unused variable on line 556 params.

File: app/services/listing_service/store/shape.rb (hunk 1 above)
Comment (rap1ds) [inline app/services/listing_service/store/shape.rb]: Will be removed in the near future.

**Pull request merged**